File: test/samples/sizedstruct/sizedstruct.py
from dataclasses import dataclass
from typing import List


@dataclass
class List_:
    """contains a list"""

    size1: int  # the list's size
    int_list: List[int]  # the integer list


@dataclass
class String:
    """contains a string"""

    size2: int  # the list's size
    string_list: str  # the string list


@dataclass
class Matrix:
    """contains a matrix"""

    size3: int  # the list's size
    list_list: List[List[int]]  # the list list


@dataclass
class NotASizedStruct:
    """this is not a 'sized struct', but a regular one!"""

    size4: int  # not the list's size
    int_list_n: List[int]  # the integer list


def sized_struct(n: int, lists: List[List_], strings: List[String], matrices: List[Matrix], same: List[NotASizedStruct]) -> None:
    """
    :param n: the size of the lists
    :param lists: a list of list of different sizes
    :param strings: a list of strings of different sizes
    :param matrices: a list of matrices of different sizes
    :param same: a list of list of same sizes
    """
    # TODO The is a special case.
    pass


if __name__ == '__main__':
    n = int(input())
    lists = [
        (lambda i: List_(
            i,
            list(map(int, input().split()))
        ))(int(input()))
        for _ in range(n)
        ]
    strings = [
        (lambda i: String(
            i,
            input()
        ))(int(input()))
        for _ in range(n)
        ]
    matrices = [
        (lambda i: Matrix(
            i,
            [list(map(int, input().split())) for _ in range(i)]
        ))(int(input()))
        for _ in range(2)
        ]
    same = [
        NotASizedStruct(
            int(input()),
            list(map(int, input().split())),
        ) for _ in range(n)]
    sized_struct(n, lists, strings, matrices, same)

File: test/samples/sizedstruct/sizedstruct.go
package main

import "bufio"
import "os"
import "strconv"
import "strings"

// contains a list
type List struct {
    size1 int // the list's size
    intList []int // the integer list
}

// contains a string
type String struct {
    size2 int // the list's size
    stringList string // the string list
}

// contains a matrix
type Matrix struct {
    size3 int // the list's size
    listList [][]int // the list list
}

// this is not a 'sized struct', but a regular one!
type NotASizedStruct struct {
    size4 int // not the list's size
    intListN []int // the integer list
}

// n: the size of the lists
// lists: a list of list of different sizes
// strings_: a list of strings of different sizes
// matrices: a list of matrices of different sizes
// same: a list of list of same sizes
func sizedStruct(n int, lists []List, strings_ []String, matrices []Matrix, same []NotASizedStruct) {
    /* TODO The is a special case. */
}

func main() {
    scanner := bufio.NewScanner(os.Stdin)
    var n int
    scanner.Scan()
    n, _ = strconv.Atoi(scanner.Text())
    lists := make([]List, n)
    for i := range lists {
        scanner.Scan()
        lists[i].size1, _ = strconv.Atoi(scanner.Text())
        lists[i].intList = make([]int, lists[i].size1)
        scanner.Scan()
        for j, jValue := range strings.Split(scanner.Text(), " ") {
            lists[i].intList[j], _ = strconv.Atoi(jValue)
        }
    }
    strings_ := make([]String, n)
    for i := range strings_ {
        scanner.Scan()
        strings_[i].size2, _ = strconv.Atoi(scanner.Text())
        scanner.Scan()
        strings_[i].stringList = scanner.Text()
    }
    matrices := make([]Matrix, 2)
    for i := range matrices {
        scanner.Scan()
        matrices[i].size3, _ = strconv.Atoi(scanner.Text())
        matrices[i].listList = make([][]int, matrices[i].size3)
        for j := range matrices[i].listList {
            matrices[i].listList[j] = make([]int, 2)
            scanner.Scan()
            for k, kValue := range strings.Split(scanner.Text(), " ") {
                matrices[i].listList[j][k], _ = strconv.Atoi(kValue)
            }
        }
    }
    same := make([]NotASizedStruct, n)
    for i := range same {
        scanner.Scan()
        same[i].size4, _ = strconv.Atoi(scanner.Text())
        same[i].intListN = make([]int, n)
        scanner.Scan()
        for j, jValue := range strings.Split(scanner.Text(), " ") {
            same[i].intListN[j], _ = strconv.Atoi(jValue)
        }
    }
    sizedStruct(n, lists, strings_, matrices, same);
}
